objects/src/testing/account_id.rs

```diff
@@ -120,3 +122,89 @@ pub const fn account_id<const CHECK_MIN_ONES: bool>(
 
     id
 }
+
+/// A builder for creating [`AccountId`]s for testing purposes.
+///
+/// This is essentially a wrapper around [`AccountId::dummy`] generating random values as its input.
+/// Refer to its documentation for details.
+///
+/// # Example
+///
+/// ```
+/// # use miden_objects::accounts::{AccountType, AccountStorageMode, AccountId};
+/// # use miden_objects::testing::account_id::{AccountIdBuilder};
+///
+/// let mut rng = rand::thread_rng();
+///
+/// // A random AccountId with random AccountType and AccountStorageMode.
+/// let random_id1: AccountId = AccountIdBuilder::new().build_with_rng(&mut rng);
+///
+/// // A random AccountId with the given AccountType and AccountStorageMode.
+/// let random_id2: AccountId = AccountIdBuilder::new()
+///     .account_type(AccountType::FungibleFaucet)
+///     .storage_mode(AccountStorageMode::Public)
+///     .build_with_rng(&mut rng);
+/// assert_eq!(random_id2.account_type(), AccountType::FungibleFaucet);
+/// assert_eq!(random_id2.storage_mode(), AccountStorageMode::Public);
+/// ```
+pub struct AccountIdBuilder {
+    account_type: Option<AccountType>,
+    storage_mode: Option<AccountStorageMode>,
+}
+
+impl AccountIdBuilder {
+    /// Creates a new [`AccountIdBuilder`].
+    pub fn new() -> Self {
+        Self { account_type: None, storage_mode: None }
+    }
```

Comment:
Should we have both `new` and `default`? I usually only implement the one (`default`).